pandas/core/algorithms.py

```diff
@@ -113,6 +113,8 @@ def isin(comps, values):
         raise TypeError("only list-like objects are allowed to be passed"
                         " to isin(), you passed a "
                         "[{0}]".format(type(values).__name__))
+    if com.is_set_like(values):
```

Comment:
instead of doing this, I think you can:

```
if not isinstance(values, np.ndarray):
    values = list(values)
```
Then don't even need the next check